CMakeLists.txt

```diff
@@ -46,18 +46,24 @@ if (NOT CMAKE_BUILD_TYPE)
             STRING "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel."
             FORCE
     )
-endif (NOT CMAKE_BUILD_TYPE)
+endif ()
 
 enable_testing()
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/)
 
+option(AUTOPAS_FORMATTING_TARGETS
+       "Adds targets for code formatting in AutoPas. Highly useful when developing AutoPas but might cause conflicts when only using it."
+       OFF)
```

Comment:
This changes the default behavior of AutoPas to not generate these targets when not explicitly told to. This is fine from a user's perspective but might be annoying from a developer's perspective. Hence I'm open to arguments to change it to `ON`.

Reply:
How about setting this variable depending on whether AutoPas is built stand-alone or inside a project?
e.g. by comparing CMAKE_SOURCE_DIR (directory of the top-level CMakeLists.txt) with AUTOPAS_SOURCE_DIR (directory of the autopas project)?
That way it would automatically be disabled if used inside of another project ;)

as this might come in handy elsewhere, we should probably define a variable AUTOPAS_STANDALONE or so, s.t., examples, tests, etc. could also be disabled automatically.

Reply:
That was a bigger idea than expected. See 46a03c5